app/screens/send-bitcoin-screen/send-bitcoin-destination-screen.tsx

```diff
@@ -111,6 +126,110 @@ const SendBitcoinDestinationScreen: React.FC<Props> = ({ route }) => {
+    fetchPolicy: "cache-and-network",
+  })
+
+  if (error) {
+    toastShow({ message: error.message, LL })
+  }
+
+  const [matchingContacts, setMatchingContacts] = useState<Contact[]>([])
+  const allContacts: Contact[] = useMemo(() => {
+    const contactsCopy = contactsData?.me?.contacts.slice() ?? []
+    const compareUsernames = (a: Contact, b: Contact) => {
+      return a.username.toLocaleLowerCase().localeCompare(b.username.toLocaleLowerCase())
+    }
+    contactsCopy.sort(compareUsernames)
+
+    return contactsCopy
+  }, [contactsData])
+
+  const [selectedId, setSelectedId] = useState("")
+
+  const handleSelection = (id: string) => {
+    if (selectedId === id) setSelectedId("")
+    else setSelectedId(id)
+  }
+
+  const reset = useCallback(() => {
+    dispatchDestinationStateAction({
+      type: "set-unparsed-destination",
+      payload: { unparsedDestination: "" },
+    })
+    setGoToNextScreenWhenValid(false)
+    setSelectedId("")
+    setMatchingContacts(allContacts)
+  }, [allContacts])
+
+  const wordMatchesContact = (searchWord: string, contact: Contact): boolean => {
+    let contactPrettyNameMatchesSearchWord: boolean
+
+    const contactNameMatchesSearchWord = contact.username
+      .toLowerCase()
+      .includes(searchWord.toLowerCase())
+
+    if (contact.alias) {
+      contactPrettyNameMatchesSearchWord = contact.alias
+        .toLowerCase()
+        .includes(searchWord.toLowerCase())
+    } else {
+      contactPrettyNameMatchesSearchWord = false
+    }
+
+    return contactNameMatchesSearchWord || contactPrettyNameMatchesSearchWord
+  }
+
+  let ListEmptyContent: React.ReactNode
+
+  if (allContacts.length > 0) {
+    ListEmptyContent = (
+      <View style={styles.emptyListNoMatching}>
+        <Text style={styles.emptyListTitle}>{LL.PeopleScreen.noMatchingContacts()}</Text>
+      </View>
+    )
+  } else if (loading) {
+    ListEmptyContent = (
+      <View style={styles.activityIndicatorContainer}>
+        <ActivityIndicator size="large" color={colors.primary} />
+      </View>
+    )
+  } else {
+    ListEmptyContent = (
+      <View style={styles.emptyListNoContacts}>
+        <Text
+          {...testProps(LL.PeopleScreen.noContactsTitle())}
+          style={styles.emptyListTitle}
+        >
+          {LL.PeopleScreen.noContactsTitle()}
+        </Text>
+        <Text style={styles.emptyListText}>{LL.PeopleScreen.noContactsYet()}</Text>
+      </View>
+    )
+  }
+
+  const updateMatchingContacts = useCallback(
+    (newSearchText: string) => {
+      if (newSearchText.length > 0) {
+        const searchWordArray = newSearchText
+          .split(" ")
+          .filter((text) => text.trim().length > 0)
+        const matchingContacts = allContacts.filter((contact) =>
+          searchWordArray.some((word) => wordMatchesContact(word, contact)),
+        )
+        setMatchingContacts(matchingContacts)
+      } else {
+        setMatchingContacts(allContacts)
+      }
+    },
```

Comment:
this function should probably also be unit tested

Reply:
Also heavily inspired from [screens/people-screen/contacts/all-contacts.tsx](https://github.com/GaloyMoney/galoy-mobile/blob/84eaa80cde72b5d8fc0ad65402066ff1ed6b72d0/app/screens/people-screen/contacts/all-contacts.tsx#L72C3-L88C4)
Let me know if revision is _really_ needed

Reply:
yes, we don't want logic in code to not be unit tested. things will break otherwise (now or on a later refactoring)